Mathlib/Analysis/SpecialFunctions/Arsinh.lean

```diff
@@ -176,6 +182,36 @@ theorem contDiff_arsinh {n : ℕ∞} : ContDiff ℝ n arsinh :=
 theorem continuous_arsinh : Continuous arsinh :=
   sinhHomeomorph.symm.continuous
 
+/-- `arcosh` is defined using a logarithm, `arcosh x = log (x + sqrt(x^2 - 1))`. -/
```

Comment:
You should add a comment about the behavior of this function outside its natural domain. Due to junk values for `√`, you get `arcosh x = log x` for `x ∈ Icc (-1) 1`. For `x < -1`, given the definition of `log` for negative values, this is `log (-x - √(x ^ 2 - 1))`.

Moreover, this actually induces the question: are you sure you want to define this function this way? Would it be more convenient to have it take different junk values outside of `Ici 1`? I'm not sure about the answer to this, but any decision you make should be explained.
```suggestion
/-- `arcosh` is defined using a logarithm, `arcosh x = log (x + √(x ^ 2 - 1))`. -/
```